extensions/libraries/redcore/api/api.php

```diff
@@ -199,9 +199,27 @@ public static function getPostedData()
 				}
 			}
 
-			// We try to transform it into XML
-			if (is_null($parsedData) && $xml = @simplexml_load_string($inputData))
+			$xml = @simplexml_load_string($inputData);
+
+			// We try to transform soap XML
+			if (is_null($parsedData) && $xml !== false)
 			{
+				/*
+					XML Namespaces suck and I couldn't get it to read the data accurately for SOAP
+					So I'm stripping everything between < and : in the xml and re-rendering it
+				    I know. I know. My skin is crawling too, but its the best I could do at the moment.
+				    If someone can make this work with namespaces that would be B-E-A-UTIFUL!
+				*/
+				$cleanData = preg_replace('/<(\/)?([a-z0-9]+):([a-z0-9]+)/i', '<$1$3', $inputData);
+				$xml = @simplexml_load_string($cleanData);
```

Comment:
what I am saying is that we are not relying on this function to get posted data in the SOAP at all... SOAP is used a bit differently than standard HTTP protocol calls, it is an RPC call. So the server actually reads and parses the envelope and passes parameters that the external client have set as parameters... as defined in the WSDL...